hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/CommandQueue.java

```diff
@@ -124,6 +127,30 @@ public int getDatanodeCommandCount(
     }
   }
 
+  /**
+   * Return a summary of all commands currently queued for the given Datanode.
+   * Note that any commands which return false for their
+   * Command.contributesToQueueSize() method will not be included in the count.
+   * At the current time, only low priority ReplicateContainerCommands meet this
+   * condition.
+   * @param datanodeUuid Datanode UUID
+   * @return A map containing the command summary. Note the returned map is a
+   *         copy of the internal map and can be modified safely by the caller.
+   */
+  public Map<SCMCommandProto.Type, Integer> getDatanodeCommandSummary(
+      final UUID datanodeUuid) {
+    lock.lock();
```

Comment:
Yea I did think about removing the commandQueue lock, especially as its a "lock" rather than "RW Lock", so its throughput would be worse than a RW lock. I was just a bit nervous about taking its lock away, but as you say, its an internal structure to the NodeManager, so provided we guard any accesses inside NodeManager it should be OK. Should we do this in a followup Jira?

Reply:
I guess a followup Jira is OK, makes it easier to revert if we are wrong. ;)

Reply:
I went ahead a made the change, as it was pretty simple. If the worst happens and we need to revert we can diff the files! But I think it will be OK - its basically double locked without this change which is just going to hurt performance.